src/backend/joanie/core/filters/admin.py

```diff
@@ -70,6 +74,43 @@ def filter_by_query(self, queryset, _name, value):
         """
         return queryset.filter(translations__title__icontains=value).distinct()
 
+    def filter_by_state(self, queryset, _name, value):
+        """
+        Filter resource by looking for states that match the one provided
+        in the "state" parameter
+        """
+        now = django_timezone.now()
+        if value == models.CourseState.ONGOING_OPEN:
+            queryset = queryset.filter(
+                start__lt=now, end__gt=now, enrollment_end__gt=now
+            )
+        elif value == models.CourseState.FUTURE_OPEN:
+            queryset = queryset.filter(start__gte=now, enrollment_end__gt=now)
```

Comment:
ditto

Reply:
Enrollment dates can be outside of course dates, so here it's necessary